glean-core/ios/Glean/Metrics/QuantityMetric.swift

```diff
@@ -0,0 +1,117 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+/// This implements the developer facing API for recording counter metrics.
+///
+/// Instances of this class type are automatically generated by the parsers at build time,
+/// allowing developers to record values that were previously registered in the metrics.yaml file.
+///
+/// The counter API only exposes the `QuantityMetricType.set(_:)` method, which takes care of validating the input
+/// data and making sure that requirements are enforced.
+public class QuantityMetricType {
+    let handle: UInt64
+    let disabled: Bool
+    let sendInPings: [String]
+
+    /// The public constructor used by automatically generated metrics.
+    public init(category: String, name: String, sendInPings: [String], lifetime: Lifetime, disabled: Bool) {
+        self.disabled = disabled
+        self.sendInPings = sendInPings
+        self.handle = withArrayOfCStrings(sendInPings) { pingArray in
+            glean_new_quantity_metric(
+                category,
+                name,
+                pingArray,
+                Int32(sendInPings.count),
+                lifetime.rawValue,
+                disabled.toByte()
+            )
+        }
+    }
+
+    /// An internal constructor to be used by the `LabeledMetricType` directly.
+    init(withHandle handle: UInt64, disabled: Bool, sendInPings: [String]) {
+        self.handle = handle
+        self.disabled = disabled
+        self.sendInPings = sendInPings
+    }
+
+    /// Destroy this metric.
+    deinit {
+        if self.handle != 0 {
+            glean_destroy_quantity_metric(self.handle)
+        }
+    }
+
+    /// Set a quantity value.
+    ///
+    /// - parameters:
+    ///     * value: The value to set. Must be non-negative.
+    public func set(_ value: Int32) {
+        guard !self.disabled else { return }
+
+        Dispatchers.shared.launchAPI {
+            glean_quantity_set(self.handle, amount)
+        }
+    }
+
+    /// Tests whether a value is stored for the metric for testing purposes only. This function will
+    /// attempt to await the last task (if any) writing to the the metric's storage engine before
+    /// returning a value.
+    ///
+    /// - parameters:
+    ///     * pingName: represents the name of the ping to retrieve the metric for.
+    ///                 Defaults to the first value in `sendInPings`.
+    /// - returns: true if metric value exists, otherwise false
+    public func testHasValue(_ pingName: String? = nil) -> Bool {
+        Dispatchers.shared.assertInTestingMode()
+
+        let pingName = pingName ?? self.sendInPings[0]
+        return glean_quantity_test_has_value(self.handle, pingName) != 0
```

Comment:
We should probably start with using `.toBool()` for this everywhere.
Right now the `!= 0` check is more common, so I'm good with keeping it here and we change that in one swoop across all metrics in a follow-up.

Reply:
https://bugzilla.mozilla.org/show_bug.cgi?id=1663469